cordapp/src/main/kotlin/com.luxoft.blockchainlab.corda.hyperledger.indy/flow/CreateClaimDefinitionFlow.kt

```diff
@@ -0,0 +1,67 @@
+package com.luxoft.blockchainlab.corda.hyperledger.indy.flow
+
+import co.paralleluniverse.fibers.Suspendable
+import com.luxoft.blockchainlab.corda.hyperledger.indy.contract.ClaimMetadataChecker
+import com.luxoft.blockchainlab.corda.hyperledger.indy.contract.SchemaChecker
+import com.luxoft.blockchainlab.corda.hyperledger.indy.data.state.IndyClaimDefinition
+import net.corda.core.contracts.Command
+import net.corda.core.contracts.StateAndContract
+import net.corda.core.flows.*
+import net.corda.core.transactions.TransactionBuilder
+
+
+/**
+ * Flow to create a credential definition and revocation registry for a schema
+ * */
+object CreateClaimDefinitionFlow {
+
+    /**
+     * @param schemaId          Id of target schema
+     * @param maxCredNumber     Maximum number of issued claims per schema
+     *
+     * @returns                 claim definition persistent id
+     * */
+    @InitiatingFlow
+    @StartableByRPC
+    class Authority(private val schemaId: String, private val maxCredNumber: Int = 100) : FlowLogic<String>() {
+
+        @Suspendable
+        override fun call(): String {
+            try {
+                val schemaFromVault = getIndySchemaState(schemaId)
+                    ?: throw RuntimeException("There is no schema with id: $schemaId in vault")
+
+                val credDef = indyUser().createClaimDefinition(schemaId, true)
```

Comment:
Need to check that credential definition already exists. If it exist and number of issued credential is less than limit (maxCredNumber) - throw CredentialDefinition already exist, otherwise if limit is rich - throw CredentialDefinition exist and need to be upgraded. If no one - create it.

Upgrade process has to be performed in separated flow - UpgradeCredentialDefinition. (can be implemented in separate PR)